File: android/app/src/main/java/com/example/app/domain/navcaddy/models/ConversationTurn.kt
package caddypro.domain.navcaddy.models

import com.google.gson.annotations.SerializedName

/**
 * Enum representing the role in a conversation turn.
 */
enum class Role {
    @SerializedName("user")
    USER,

    @SerializedName("assistant")
    ASSISTANT
}

/**
 * Domain model representing a single turn in the conversation history.
 *
 * Used for maintaining context across multiple user inputs.
 *
 * Spec reference: navcaddy-engine.md R6
 *
 * @property role Who spoke in this turn
 * @property content The message content
 * @property timestamp Unix timestamp of the turn
 */
data class ConversationTurn(
    @SerializedName("role")
    val role: Role,

    @SerializedName("content")
    val content: String,

    @SerializedName("timestamp")
    val timestamp: Long
)

File: ios/App/Sources/Core/Domain/NavCaddy/ConversationTurn.swift
import Foundation

/// Represents a single turn in the conversation history.
struct ConversationTurn: Codable, Hashable, Identifiable {
    enum Role: String, Codable, Hashable {
        case user
        case assistant
    }

    let id: String
    let role: Role
    let content: String
    let timestamp: Date

    init(
        id: String = UUID().uuidString,
        role: Role,
        content: String,
        timestamp: Date = Date()
    ) {
        self.id = id
        self.role = role
        self.content = content
        self.timestamp = timestamp
    }
}
